puppet/custom_device_context.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python
+
+import argparse
+import json
+import requests
+
+from requests.auth import HTTPBasicAuth
+from typing import List
+
+SOURCE = "puppet"
+
+
+def read_data_file(file: str) -> List:
+    f = open(file)
+    data = json.load(f)
+    f.close()
+    return data
+
+
+# Parses report and push to devicecontext API
+def map_and_call_device_context_api(data: List, client_id: str, secret_key: str):
+    for record in data:
+        for _, device in record.items():
+            device_details = device["facts"]
+            api_payload = {
+                "mac_address": device_details["macaddress"],
+                "ip": device_details["ipaddress"],
+                "source": SOURCE,
+                "hostname": device_details["hostname"],
+                "manufacturer": device_details["manufacturer"],
+                "os": device_details.get("os", {}).get("name"),
+                "username": device_details.get("identity", {}).get("user"),
+                "use_asset": True
```

Comment:
If you define a `Dataclass` then all this will change to attribute-style instead of key-based look up.